dasharo-compatibility/usb-type-c-semiauto.robot

```diff
@@ -0,0 +1,607 @@
+    Pause Execution In Console    Please make sure the docking station is disconnected and press ENTER
+    Set Platform Sleep Type    S3
+    Docking Station Detection After Suspend Then Hotplug (Ubuntu 22.04)    WL-UG69PD2 Rev.A1    S3
+
+
+*** Keywords ***
+Docking Station Detection After Coldboot (Ubuntu 22.04)
+    [Arguments]    ${docking_station_model}
+    Pause Execution In Console
+    ...    Please make sure the docking station connected is ${docking_station_model} and press ENTER
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${out_after_reboot}=    Execute Linux Command    uptime --since
+    Detect Docking Station In Linux    ${docking_station_model}
+    Set Global Variable    ${FAILED_DETECTION}    0
+    FOR    ${iteration}    IN RANGE    0    ${STABILITY_DETECTION_COLDBOOT_ITERATIONS}
+        Log To Console    Cold boot iteration ${iteration+1}/${STABILITY_DETECTION_COLDBOOT_ITERATIONS}
+        TRY
+            ${out_before_reboot}=    Execute Linux Command    uptime --since
+            WHILE    '${out_before_reboot}' == '${out_after_reboot}'
+                Log To Console    Coldboot the DUT manually
+                # coldboot - msi ./sonoff, protectli RteCtrl -rel, novacustom ???
+                Pause Execution In Console    Do power cut-off, reconnect and press ENTER.
+                Login To Linux
+                Switch To Root User
+                ${out_after_reboot}=    Execute Linux Command    uptime --since
+                Detect Docking Station In Linux    ${docking_station_model}
+            END
+        EXCEPT
+            ${failed_detection}=    Evaluate    ${FAILED_DETECTION} + 1
+        END
+    END
+    IF    '${failed_detection}' > '${ALLOWED_DOCKING_STATION_DETECT_FAILS}'
+        FAIL    \n ${failed_detection} iterations failed.
+    END
+    Log To Console    \nAll iterations passed.
+
+Docking Station Detection After Warmboot (Ubuntu 22.04)
+    [Arguments]    ${docking_station_model}
+    Pause Execution In Console
+    ...    Please make sure the docking station connected is ${docking_station_model} and press ENTER
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${out_after_reboot}=    Execute Linux Command    uptime --since
+    Detect Docking Station In Linux    ${docking_station_model}
+    Set Global Variable    ${FAILED_DETECTION}    0
+    FOR    ${iteration}    IN RANGE    0    ${STABILITY_DETECTION_WARMBOOT_ITERATIONS}
+        Log To Console    Warm boot iteration ${iteration+1}/${STABILITY_DETECTION_WARMBOOT_ITERATIONS}
+        TRY
+            ${out_before_reboot}=    Execute Linux Command    uptime --since
+            WHILE    '${out_before_reboot}' == '${out_after_reboot}'
+                Write Into Terminal    poweroff
+                Log To Console    Warmboot the DUT manually
+                # warmboot - msi rte, protectli novacustom ???
```

Comment:
Warmboot is boot from S5, coldboot is boot from G3.

- When laptop is off, it's by default in G3
- When you power it on, it goes to S5->S3->S0 (working state)
- When you power it off, **if AC adapter is connected** it'll stay in S5, if you disconnect power, it'll go to G3 again.

So testing warmboot will be difficult with our current tests turning laptop on via Power on AC... need to add another method for the power-on signal, like WoL.